sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -850,7 +889,22 @@ private async Task RunAsync(CancellationToken cancellationToken)
 
                 await partitionPump.StartAsync().ConfigureAwait(false);
 
-                PartitionPumps[partitionId] = partitionPump;
+                PartitionPump existingPump = null;
+                PartitionPumps.AddOrUpdate(
+                    partitionId,
+                    partitionPump,
+                    (partitionID, existingValue) =>
+                    {
+                        existingPump = existingValue;
+                        return partitionPump;
+                    });
+
+                //stop the pump we replaced
+                if (existingPump != null)
+                {
+                    await StopPartitionPumpAsync(partitionId, ProcessingStoppedReason.Shutdown, existingPump);
```

Comment:
I believe we're missing a `ConfigureAwait(false)` here.

Reply:
thanks!